From 764d4cfb81c377abbca691fb6417c83553037542 Mon Sep 17 00:00:00 2001
From: Shahed Shaikh <shahedx.shaikh@intel.com>
Date: Wed, 15 Jul 2020 17:17:03 +0800
Subject: [PATCH] DRV_DPM_SW-328: DPM support Queue level PIR

Convert PIR passed by caller to EIR for QOS queue conf
using below formula
EIR = PIR - CIR
where
EIR = Excess information rate
PIR = Peak information rate
CIR = Committed information rate

Signed-off-by: Shahed Shaikh <shahedx.shaikh@intel.com>
---
 drivers/net/datapath/dpm/gswip31/datapath_ppv4_api.c | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/net/datapath/dpm/gswip31/datapath_ppv4_api.c b/drivers/net/datapath/dpm/gswip31/datapath_ppv4_api.c
index 58880e3d3a4d..568010a0bb96 100644
--- a/drivers/net/datapath/dpm/gswip31/datapath_ppv4_api.c
+++ b/drivers/net/datapath/dpm/gswip31/datapath_ppv4_api.c
@@ -3568,6 +3568,7 @@ int dp_shaper_conf_set_31(struct dp_shaper_conf *cfg, int flag)
 				return DP_FAILURE;
 			}
 			queue_cfg.common_prop.bandwidth_limit = bw_limit;
+			queue_cfg.eir = cfg->pir > cfg->cir ? cfg->pir - cfg->cir : 0;
 			break;
 		case DP_SHAPER_CMD_REMOVE:
 		case DP_SHAPER_CMD_DISABLE:
@@ -3677,7 +3678,7 @@ int dp_shaper_conf_get_31(struct dp_shaper_conf *cfg, int flag)
 	struct pp_qos_port_conf port_cfg = {0};
 	struct hal_priv *priv;
 	int node_id, res;
-	u32 bw_limit;
+	u32 bw_limit, pir = 0;
 
 	if (!cfg) {
 		pr_err("cfg cannot be NULL\n");
@@ -3710,6 +3711,8 @@ int dp_shaper_conf_get_31(struct dp_shaper_conf *cfg, int flag)
 			       queue_cfg.common_prop.bandwidth_limit);
 			return DP_FAILURE;
 		}
+
+		pir = queue_cfg.common_prop.bandwidth_limit + queue_cfg.eir;
 	} else if (cfg->type == DP_NODE_SCH) {
 
 		if (!is_sch_valid(cfg->id.sch_id) ||
@@ -3756,7 +3759,7 @@ int dp_shaper_conf_get_31(struct dp_shaper_conf *cfg, int flag)
 	}
 
 	cfg->cir = bw_limit;
-	cfg->pir = 0;
+	cfg->pir = pir;
 	cfg->cbs = 0;
 	cfg->pbs = 0;
 	return DP_SUCCESS;
